### PR TITLE
Open-wc button tests

### PR DESCRIPTION
**Pull request by dlockhart**
This converts the remaining button tests to use the open-wc testing tools.

Let me know if this is too much to review at once... a single test seemed too little, but there were a lot of buttons things. The other components don't have as many nested components so should be easier.

### CODE REVIEW

File: components/button/test/button-icon.test.js [hunk 1]
```diff
@@ -1,31 +1,32 @@
 import '../button-icon.js';
```
Comment (dlockhart) [inline components/button/test/button-icon.test.js]: Just a bit of cleanup in here as I learned some new things.

File: README.md [hunk 2]
```diff
@@ -113,7 +113,7 @@ npm run build && npm run test:diff:golden && npm run test:diff:golden:commit
 To run local unit tests:
 
 ```shell
-npm run test:local
+npm run test:headless
```
Comment (dlockhart) [inline README.md]: Missed this in a previous PR.

File: components/button/test/button-mixin.test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+const tagName = defineCE(
+	class extends ButtonMixin(LitElement) {
+		render() {
+			return html`
+				<button>Test Button</button>
+			`;
+		}
+	}
+);
```
Comment (dlockhart) [inline components/button/test/button-mixin.test.js]: This is pretty sweet -- we can define elements inline so we hopefully don't need as many test/demo components.

File: components/button/test/floating-buttons.html [hunk 4]
```diff
@@ -1,191 +0,0 @@
-<!doctype html>
```
Comment (dlockhart) [inline components/button/test/floating-buttons.html]: I deleted these tests as they had a huge amount of overlap with the visual diff tests and seemed unnecessary.

File: package.json [hunk 5]
```diff
@@ -21,6 +21,7 @@
     "test:diff:golden:commit": "commit-goldens",
     "test:headless": "npm run test:headless:polymer && npm run test:headless:karma",
     "test:headless:karma": "karma start",
+    "test:headless:karma:watch": "karma start --auto-watch=true --single-run=false",
```
Comment (dlockhart) [inline package.json]: Helpful for debugging.

**Comment by margaree** [comment]
Should the .eslintrc.json file in `button/test` be switched to use the `open-wc-testing-config` one?

**Comment by dlockhart** [comment]
Yes, thanks @margaree!

**Pull request merged**